src/main/java/teammates/common/datatransfer/FeedbackConstantSumQuestionDetails.java

```diff
@@ -450,14 +474,41 @@ public String getQuestionResultStatisticsCsv(
 
         DecimalFormat df = new DecimalFormat("#.##");
         
+        Map<String, List<Integer>> sortedOptionPoints = new TreeMap<String, List<Integer>>();
+        
+        Map<String, String> identifierMap = new HashMap<String, String>();
+        
         for (Entry<String, List<Integer>> entry : optionPoints.entrySet()) {
+            
+            if (distributeToRecipients) {
+                
+                String participantIdentifier = entry.getKey();
+                String name = bundle.getNameForEmail(participantIdentifier);
+                String nameEmail = name + participantIdentifier;
+                
+                identifierMap.put(nameEmail, participantIdentifier);
+                sortedOptionPoints.put(nameEmail, entry.getValue());
+                
+            } else {
+                
+                String option = options.get(Integer.parseInt(entry.getKey()));
+                
+                sortedOptionPoints.put(option, entry.getValue());
+            }
+        }
```

Comment:
Extracted 2 common helper methods into the current java file, still rather specific to the current class.